### ISSUE TITLE
Add method zipWith(Iterable, BiFunction) as shortcut for zip(Iterable).map(Function)

### ISSUE THREAD

**Issue by danieldietrich** [vavr-io/vavr#1313]
We should add `zipWith` to all types that already have `zip`.

Here is an example of an implementation in Stream. Because Stream is lazy evaluated, other types that are eager evaluated may have another implementation.

``` java
default <U, R> Stream<R> zipWith(Iterable<? extends U> iterable, BiFunction<? super T, ? super U, ? extends R> combiner) {
    return zip(iterable).map(t -> combiner.apply(t._1, t._2));
}
```

**Comment by danieldietrich** [comment]
**Note:** It would be more _functional_ to switch the arguments if we had curried argument lists in Java (like in Scala):

```java
// zip = stream1.zipWith(f); zipped = zip.apply(stream2);
default <U, R> Stream<R> zipWith(BiFunction<? super T, ? super U, ? extends R> combiner)(Iterable<? extends U> iterable) {
    ...
}
```

or have a static method to obtain a curried form:

```java
// zip = zipWith(f); zipped = zip.apply(stream1, stream2);
static <T, U, R> BiFunction<Stream<T>, Stream<U>, Stream<R>> zipWith(BiFunction<? super T, ? super U, ? extends R> combiner) {
    ...
}
```

But since Java is an object-oriented language, this seems more natural:

```java
// = stream1.zip(stream2).map(t -> f.apply(t._1, t._2))
stream1.zipWith(stream2, f);
```

**Comment by valery1707** [comment]
This issue was implemented by #1427 isn't it?

**Comment by danieldietrich** [comment]
Yes, thx, will close it.
